nessie/system_test.go

```diff
@@ -110,8 +137,8 @@ func uploadContent(ctx context.Context, repo string, branch string, objPath stri
 	}, w.FormDataContentType(), &b)
 }
 
-func uploadFileRandomData(ctx context.Context, t *testing.T, repo, branch, objPath string) (checksum, content string) {
-	checksum, content, err := uploadFileRandomDataAndReport(ctx, repo, branch, objPath)
+func uploadFileRandomData(ctx context.Context, t *testing.T, repo, branch, objPath string, direct bool) (checksum, content string) {
+	checksum, content, err := uploadFileRandomDataAndReport(ctx, t, repo, branch, objPath, direct && directUploadSupported)
```

Comment:
Maybe check the value of `directUploadSupported` in the inner method instead of maybe missing it occasionally from other callers of `uploadFileRandomDataAndReport`.

Reply:
This no longer exists, and `direct` is piped right through.